ykrt/src/mt.rs

```diff
@@ -85,6 +85,11 @@ pub struct MT {
     /// How many worker threads are currently running. Note that this may temporarily be `>`
     /// [`max_worker_threads`].
     active_worker_threads: AtomicUsize,
+    #[cfg(yk_llvm_sync_hack)]
+    /// A temporary hack which keeps track of how many compile (etc.) jobs are active, because LLVM
+    /// dies horribly if the main thread exits before those jobs are finished. This then marries up
+    /// with Self::.llvm_sync_hack().
+    active_worker_jobs: AtomicUsize,
```

Comment:
Did you opt for atomics because they are cheaper than stashing join handles?

Reply:
Join handles would be harder, because worker threads aren't necessarily active, so "join handles" would require a) a condvar call to wake them up b) a bool to let them know to exit. So the horrors of an (atomic) count are at least briefer horrrors.

Reply:
> because worker threads aren't necessarily active

Dumb question: what do we mean by an "active thread"?

Perhaps a worker thread doesn't exit after it has performed a task? It hangs around waiting for a new task?

Reply:
Worker threads wait for jobs: so "active" means "a worker thread that's actively doing something", excluding "worker threads waiting for a job to do".

Reply:
Right. Now I understand your prior comment. Threads don't join after a compilation task finishes, so a join handle doesn't really cut it.